### DISCUSSION TITLE
Offline Thumbnail Regeneration

### DISCUSSION THREAD

**Discussion by MikNolhan** [immich-app/immich#12560] [Q&A]
Question 1: When clicking the button to generate all thumbnails, does immich regenerate and replace/remove, or are the old ones left behind and tracked, but just irrelevant? Would doing this cause everything to just grow in excess? From what I understand the answer is that it will regenerate and update the hash in postgres to make the connection.

Question 2: If thumbnails for whatever the reason are reported as offline, and the button is used to generate them all again, will they no longer be reported offline once those thumbnails have been generated for those asset IDs?

Question 3: Is there a way to select multiple photos in ANY view, and without having to go into each photo, quickly regenerate thumbnails for those specifically?

**Comment by bo0tzz** [comment]
1. Old thumbnails should get cleaned up
2. If Immich generates a new thumbnail then that should no longer report as offline, yeah
3. Multi-select and the menu options aren't yet unified across all the views, so no.
